src/Microsoft.AspNet.Owin/OwinEnvironment.cs

```diff
@@ -61,7 +61,7 @@ public OwinEnvironment(HttpContext context)
                 { OwinConstants.ResponseReasonPhrase, new FeatureMap<IHttpResponseFeature>(feature => feature.ReasonPhrase, (feature, value) => feature.ReasonPhrase = Convert.ToString(value)) },
                 { OwinConstants.ResponseHeaders, new FeatureMap<IHttpResponseFeature>(feature => feature.Headers, (feature, value) => feature.Headers = (IDictionary<string, string[]>)value) },
                 { OwinConstants.ResponseBody, new FeatureMap<IHttpResponseFeature>(feature => feature.Body, () => Stream.Null, (feature, value) => feature.Body = (Stream)value) },
-                { OwinConstants.CommonKeys.OnSendingHeaders, new FeatureMap<IHttpResponseFeature>(feature => new Action<Action<object>, object>(feature.OnResponseStarting)) },
+                { OwinConstants.CommonKeys.OnSendingHeaders, new FeatureMap<IHttpResponseFeature>(feature => new Action<Func<object, Task>, object>(feature.OnResponseStarting)) },
```

Comment:
Yeah, this will require some adaptation. OWIN middleware on ASP.NET servers will be easy because we can wrap an `Action<object>` in a `Func<object, Task>`. However, running ASP.NET components on OWIN servers will be a little awkward because we have to wrap a  `Func<object, Task>` in a `Action<object>` and block.